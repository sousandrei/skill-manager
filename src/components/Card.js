import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { delSkill } from '../actions/skills'

export class Card extends React.Component {

	render() {
		const {
			index,
			delSkill,
			skill: {
				id,
				name,
				experience
			}
		} = this.props

		return (
			<div>
				<span>{index}</span>
				<span>{name}</span>
				<span>{experience}</span>
				<button type='button'
					onClick={() => delSkill(id)}>x</button>
			</div>
		)
	}
}

Card.propTypes = {
	index: PropTypes.number,
	skill: PropTypes.object,
	delSkill: PropTypes.func
}

const mapDispatchToProps = dispatch => ({
	delSkill: id => dispatch(delSkill(id))
})

export default connect(undefined, mapDispatchToProps)(Card)