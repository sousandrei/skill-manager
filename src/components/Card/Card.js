import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { delSkill } from '../../actions/skills/skills'

export class Card extends React.Component {

	render() {
		const {
			index,
			delSkill,
			skill: {
				id,
				name,
				experience
			}
		} = this.props

		return (
			<div className='card'>
				<span className='card__index'>{index + 1}</span>
				<div className='card__text'>
					<span className='card__name'>{name}</span>
					<span className='card__exp'>{experience}</span>
				</div>
				<button
					className='card__button'
					type='button'
					onClick={() => delSkill(id)}>
					<i className='fas fa-times'></i>
				</button>
			</div>
		)
	}
}

Card.propTypes = {
	index: PropTypes.number,
	skill: PropTypes.object,
	delSkill: PropTypes.func
}

const mapDispatchToProps = dispatch => ({
	delSkill: id => dispatch(delSkill(id))
})

export default connect(undefined, mapDispatchToProps)(Card)