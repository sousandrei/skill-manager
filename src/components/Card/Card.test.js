import React from 'react'
import { shallow } from 'enzyme'

import { Card } from './Card'
import SkillMock from '../../mocks/Skill.mock'

test('should render card', () => {

	const wrapper = shallow(<Card
		index={1}
		skill={SkillMock} />)

	expect(wrapper).toMatchSnapshot()
})

