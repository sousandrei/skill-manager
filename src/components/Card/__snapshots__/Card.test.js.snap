// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render card 1`] = `
<div
  className="card"
>
  <span
    className="card__index"
  >
    2
  </span>
  <div
    className="card__text"
  >
    <span
      className="card__name"
    >
      skill-1
    </span>
    <span
      className="card__exp"
    >
      1 year
    </span>
  </div>
  <button
    className="card__button"
    onClick={[Function]}
    type="button"
  >
    <i
      className="fas fa-times"
    />
  </button>
</div>
`;
